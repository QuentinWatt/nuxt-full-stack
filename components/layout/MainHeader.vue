<template>
  <header class="bg-white border-b">
    <div class="container mx-auto px-3 h-16 flex items-center justify-between">
      <span class="font-bold text-3xl">MyLogo</span>
      <nav>
        <ul class="flex items-center">
          <li><NuxtLink to="/" class="mr-2">Home</NuxtLink></li>
          <li><NuxtLink to="/about">About</NuxtLink></li>
        </ul>
      </nav>
    </div>
  </header>
</template>