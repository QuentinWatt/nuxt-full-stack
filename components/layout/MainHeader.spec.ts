import { expect, test } from "vitest";
import MainHeader from "./MainHeader.vue";
import { mountSuspended } from "@nuxt/test-utils/runtime";

test("It matches the snapshot", async () => {
  const component = await mountSuspended(MainHeader);
  expect(component.html()).matchSnapshot();
});

test("It has a logo", async () => {
  const component = await mountSuspended(MainHeader);
  expect(component.html()).toContain("MyLogo");
});

test("It has the correct menu items", async () => {
  const component = await mountSuspended(MainHeader);
  expect(component.html()).toContain("Home");
  expect(component.html()).toContain("About");
});
