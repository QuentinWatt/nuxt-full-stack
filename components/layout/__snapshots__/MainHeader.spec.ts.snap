// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`It matches the snapshot 1`] = `
"<header class="bg-white border-b">
  <div class="container mx-auto px-3 h-16 flex items-center justify-between"><span class="font-bold text-3xl">MyLogo</span>
    <nav>
      <ul class="flex items-center">
        <li><a href="/" class="mr-2">Home</a></li>
        <li><a href="/about">About</a></li>
      </ul>
    </nav>
  </div>
</header>"
`;
